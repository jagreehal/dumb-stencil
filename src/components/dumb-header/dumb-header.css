:host {
  display: grid;
}

.c-header {
  background: #ddd;
  display: inherit;
  gap: 10px;
  grid-auto-flow: column;
  grid-template-columns: 1fr;
  padding: 1rem 2rem;
}
