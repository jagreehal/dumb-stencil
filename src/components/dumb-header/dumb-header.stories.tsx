import '@stencil/core';

import { storiesOf } from '@storybook/html';

const stories = storiesOf('Global/Header', module);

stories.add('Default', () => {
  return `<dumb-header></dumb-header>`;
});
