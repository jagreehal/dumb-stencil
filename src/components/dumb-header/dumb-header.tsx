import { Component, Prop } from '@stencil/core';
import { PrimaryNavItem } from '../dumb-primary-nav/dumb-primary-nav';

const listItems: PrimaryNavItem[] = [
  {
    href: '#',
    text: 'Nav Item 1'
  },
  {
    href: '#',
    text: 'Nav Item 2'
  },
  {
    href: '#',
    text: 'Nav Item 3'
  },
  {
    href: '#',
    text: 'Nav Item 4'
  }
];
@Component({
  tag: 'dumb-header',
  styleUrl: 'dumb-header.css',
  shadow: true
})
export class DumbHeader {
  @Prop() companyName: string;
  @Prop() listItems: string;

  render() {
    return (
      <header class="c-header">
        <dumb-logo
          href="#"
          src="http://via.placeholder.com/120x60"
          alt={this.companyName}
        />
        <dumb-primary-nav listItems={listItems} />
        <dumb-inline-form
          method="post"
          action="#"
          placeholder="Search this site"
          cta="Search"
          id="search-form"
          label="Search the site"
        />
      </header>
    );
  }
}
