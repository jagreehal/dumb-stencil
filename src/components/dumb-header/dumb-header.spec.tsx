import { TestWindow } from '@stencil/core/testing';
import { DumbHeader } from './dumb-header';

describe('dumb-header', () => {
  let element;
  let window;
  beforeEach(async () => {
    window = new TestWindow();
    element = await window.load({
      components: [DumbHeader],
      html: `<dumb-header></dumb-header>`
    });
  });

  it('should match snapshot', async () => {
    expect(element).toMatchSnapshot();
  });
});
