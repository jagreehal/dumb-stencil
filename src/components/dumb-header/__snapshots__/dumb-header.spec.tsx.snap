// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dumb-header should match snapshot 1`] = `
<dumb-header
  class="hydrated"
>
  
  <header
    class="c-header"
  >
    <dumb-logo
      href="#"
      src="http://via.placeholder.com/120x60"
    />
    <dumb-primary-nav />
    <dumb-inline-form
      action="#"
      cta="Search"
      id="search-form"
      label="Search the site"
      method="post"
      placeholder="Search this site"
    />
  </header>
</dumb-header>
`;
